src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -306,6 +310,20 @@ private function registerBusMiddleware(ContainerBuilder $container, string $busI
                 throw new RuntimeException(sprintf('Invalid middleware factory "%s": a middleware factory must be an abstract definition.', $id));
             }
 
+            if ($debug) {
+                $container->register($debugMiddlewareId = "debug.traced.$messengerMiddlewareId", TraceableMiddleware::class)
+                    // Decorates with a high priority so it's applied the earliest:
+                    ->setDecoratedService($messengerMiddlewareId, null, 100)
+                    ->setArguments(array(
+                        new Reference("$debugMiddlewareId.inner"),
+                        new Reference($this->debugStopwatch),
+                        // If definition isn't abstract, we can't be sure the service instance
+                        // is only used in one bus at the same time, so we don't inject the bus name in this case:
+                        $isAbstract ? $busId : null,
```

Comment:
the implementation seems to do exactly the opposite of the comment.

Reply:
> If definition isn't abstract [...] we don't inject the bus name

Feels like it is to me. But perhaps can be reworded.

Reply:
but here, you do inject it *only* when it is abstract